new-lamassu-admin/src/pages/Maintenance/CashboxHistory.js

```diff
@@ -0,0 +1,267 @@
+import { useQuery, useMutation } from '@apollo/react-hooks'
+import { makeStyles } from '@material-ui/core'
+import gql from 'graphql-tag'
+import moment from 'moment'
+import * as R from 'ramda'
+import React, { useState } from 'react'
+import * as Yup from 'yup'
+
+import { Link, IconButton } from 'src/components/buttons'
+import { TextInput } from 'src/components/inputs'
+import { NumberInput } from 'src/components/inputs/formik'
+import DataTable from 'src/components/tables/DataTable'
+import { ReactComponent as EditIcon } from 'src/styling/icons/action/edit/enabled.svg'
+import { ReactComponent as TxInIcon } from 'src/styling/icons/direction/cash-in.svg'
+import { ReactComponent as TxOutIcon } from 'src/styling/icons/direction/cash-out.svg'
+
+const GET_BATCHES = gql`
+  query cashboxBatches {
+    cashboxBatches {
+      id
+      deviceId
+      created
+      operationType
+      customBillCount
+      performedBy
+      bills {
+        fiat
+        deviceId
+        created
+        cashbox
+      }
+    }
+  }
+`
+
+const EDIT_BATCH = gql`
+  mutation editBatch($id: ID, $performedBy: String) {
+    editBatch(id: $id, performedBy: $performedBy) {
+      id
+    }
+  }
+`
+
+const styles = {
+  operationType: {
+    marginLeft: 8
+  },
+  operationTypeWrapper: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center'
+  },
+  saveAndCancel: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between'
+  }
+}
+
+const schema = Yup.object().shape({
+  performedBy: Yup.string().nullable()
+})
+
+const useStyles = makeStyles(styles)
+
+const CashboxHistory = ({ machines, currency }) => {
+  const classes = useStyles()
+  const [error, setError] = useState(false)
+  const [fields, setFields] = useState([])
+
+  const { data, loading } = useQuery(GET_BATCHES)
+
+  const [editBatch] = useMutation(EDIT_BATCH, {
+    refetchQueries: () => ['cashboxBatches']
+  })
+
+  const batches = R.path(['cashboxBatches'])(data)
+
+  const getOperationRender = {
+    'cash-in-empty': (
+      <>
+        <TxInIcon />
+        <span className={classes.operationType}>Cash-in emptied</span>
+      </>
+    ),
+    'cash-out-1-refill': (
+      <>
+        <TxOutIcon />
+        <span className={classes.operationType}>Cash-out 1 refill</span>
+      </>
+    ),
+    'cash-out-1-empty': (
+      <>
+        <TxOutIcon />
+        <span className={classes.operationType}>Cash-out 1 emptied</span>
+      </>
+    ),
+    'cash-out-2-refill': (
+      <>
+        <TxOutIcon />
+        <span className={classes.operationType}>Cash-out 2 refill</span>
+      </>
+    ),
+    'cash-out-2-empty': (
+      <>
+        <TxOutIcon />
+        <span className={classes.operationType}>Cash-out 2 emptied</span>
+      </>
+    )
+  }
```

Comment:
Same here. A refactor should be needed to transform this into a function where the cassetteNumber is received, to avoid having multiple lines repeated where the only thing that changes is a number